tket/src/Mapping/MappingFrontier.cpp

```diff
@@ -591,6 +583,18 @@ bool MappingFrontier::add_swap(const UnitID& uid_0, const UnitID& uid_1) {
   bool uid1_ancilla =
       this->ancilla_nodes_.find(n1) != this->ancilla_nodes_.end();
 
+  // In this case we can still use the node, but we need to make sure that
+  // the wire it's on won't be reassigned
+  // This could reduce performance, but the "reassignable_nodes_" case
+  // deals with Qubit wires that have no multi-qubit quantum operations,
+  // so in practice this won't change the results for a meaningful circuit
+  if (this->reassignable_nodes_.find(n0) != this->reassignable_nodes_.end()) {
+    this->reassignable_nodes_.erase(n0);
```

Comment:
Can we try to exercise this condition in a test?

Reply:
done